app/models/user.rb

```diff
@@ -0,0 +1,29 @@
+class User < ApplicationRecord
+  before_save{:email_downcase}
+
+  validates :name, presence: true,
+    length: {maximum: Settings.length.name}
+
+  validates :email, presence: true,
+    length: {maximum: Settings.length.email},
+    format: {with: Settings.regex.email_regex},
+    uniqueness: true
+
+  has_secure_password
+  validates :password, presence: true,
+    length: {minimum: Settings.length.password}
+  validate :valid_birthday
+
+  private
+
+  def valid_birthday
+    if birthday.present?
+      return if Date.parse(birthday) > Settings.length.date.year.ago
+    end
```

Comment:
```suggestion
    return if Date.parse(birthday) > Settings.length.date.year.ago
```

chỗ gọi method này thì ` validate :valid_birthday, if -> {birthday.present?}`